Cannot cross-compile for Solaris

When trying to build a project that depends on starlark-go with `GOOS=solaris`, compiler throws the following error:

```
/go/pkg/mod/go.starlark.net@v0.0.0-20201014215153-dff0ae5b4820/starlark/int_posix64.go:70:12: undefined: syscall.Mmap
```

Is there a reason why `golang.org/x/sys` (which would otherwise provide the `Mmap()` implementation) was ditched in favor to a in-house implementation in https://github.com/google/starlark-go/pull/287?

--- a/go.mod
+++ b/go.mod
@@ -6,5 +6,5 @@ require (
 	github.com/chzyer/logex v1.1.10 // indirect
 	github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e
 	github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1 // indirect
-	golang.org/x/sys v0.0.0-20200803210538-64077c9b5642 // indirect
+	golang.org/x/sys v0.0.0-20200803210538-64077c9b5642
 )

--- a/starlark/eval_test.go
+++ b/starlark/eval_test.go
@@ -8,6 +8,7 @@ import (
 	"bytes"
 	"fmt"
 	"math"
+	"os/exec"
 	"path/filepath"
 	"sort"
 	"strings"
@@ -888,5 +889,27 @@ func TestExecutionSteps(t *testing.T) {
 	_, err = countSteps(1000)
 	if fmt.Sprint(err) != "Starlark computation cancelled: too many steps" {
 		t.Errorf("execution returned error %q, want cancellation", err)
+	}
+}
+
+// TestDeps fails if the interpreter proper (not the REPL, etc) sprouts new external dependencies.
+// We may expand the list of permitted dependencies, but should do so deliberately, not casually.
+func TestDeps(t *testing.T) {
+	cmd := exec.Command("go", "list", "-deps")
+	out, err := cmd.Output()
+	if err != nil {
+		t.Skipf("'go list' failed: %s", err)
+	}
+	for _, pkg := range strings.Split(string(out), "\n") {
+		// Does pkg have form "domain.name/dir"?
+		slash := strings.IndexByte(pkg, '/')
+		dot := strings.IndexByte(pkg, '.')
+		if 0 < dot && dot < slash {
+			if strings.HasPrefix(pkg, "go.starlark.net/") ||
+				strings.HasPrefix(pkg, "golang.org/x/sys/") {
+				continue // permitted dependencies
+			}
+			t.Errorf("new interpreter dependency: %s", pkg)
+		}
 	}
 }

--- a/starlark/int_posix64.go
+++ b/starlark/int_posix64.go
@@ -22,9 +22,9 @@ import (
 	"log"
 	"math"
 	"math/big"
-	"runtime"
-	"syscall"
 	"unsafe"
+
+	"golang.org/x/sys/unix"
 )
 
 // intImpl represents a union of (int32, *big.Int) in a single pointer,
@@ -59,15 +59,7 @@ func makeBigInt(x *big.Int) Int { return Int{intImpl(x)} }
 var smallints = reserveAddresses(1 << 32)
 
 func reserveAddresses(len int) uintptr {
-	// Use syscall to avoid golang.org/x/sys/unix dependency.
-	MAP_ANON := 0x1000 // darwin (and all BSDs)
-	switch runtime.GOOS {
-	case "linux", "android":
-		MAP_ANON = 0x20
-	case "solaris":
-		MAP_ANON = 0x100
-	}
-	b, err := syscall.Mmap(-1, 0, len, syscall.PROT_READ, syscall.MAP_PRIVATE|MAP_ANON)
+	b, err := unix.Mmap(-1, 0, len, unix.PROT_READ, unix.MAP_PRIVATE|unix.MAP_ANON)
 	if err != nil {
 		log.Fatalf("mmap: %v", err)
 	}

